prov/efa/src/rxr/rxr_pkt_entry.c

```diff
@@ -362,13 +362,18 @@ ssize_t rxr_pkt_entry_sendv(struct rxr_ep *ep,
 	struct rxr_peer *peer;
 
 	msg.msg_iov = iov;
+	msg.addr = addr;
 	msg.desc = desc;
 	msg.iov_count = count;
 	peer = rxr_ep_get_peer(ep, addr);
-	msg.addr = (peer->is_local) ? peer->shm_fiaddr : addr;
 	msg.context = pkt_entry;
 	msg.data = 0;
 
+	if (peer->is_local) {
```

Comment:
Perhaps this should be in `rxr_pkt_entry_sendmsg`, since that's where fi_sendmsg is called? Looks like nothing calls sendmsg directly at the moment but that could change.

Reply:
Is it possible to make this change? Otherwise it looks fine to me.

Reply:
Move it to `rxr_pkt_entry_sendmsg` would require changing the function's signature by dropping the `const` prefix to msg, because we will need to modify `msg->desc` inside the function. I am open to it, but wonder if you are OK with that?

Reply:
Oh, I see. In that case, I'd just add a comment to `rxr_pkt_entry_sendmsg` that the caller must set `msg->desc` correctly based on whether this going over SHM, and to see sendv for an example.